code/datums/components/magnetic_catch.dm

```diff
@@ -1,23 +1,34 @@
 /datum/component/magnetic_catch/Initialize()
 	if(!isatom(parent))
 		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine)
 	if(ismovableatom(parent))
-		RegisterSignal(parent, COMSIG_MOVABLE_UNCROSS, .proc/uncross_react)
+		RegisterSignal(parent, COMSIG_MOVABLE_CROSSED, .proc/crossed_react)
+		RegisterSignal(parent, COMSIG_MOVABLE_UNCROSSED, .proc/uncrossed_react)
+		for(var/i in get_turf(parent))
+			if(i == parent)
+				continue
+			RegisterSignal(i, COMSIG_MOVABLE_PRE_THROW, .proc/throw_react)
 	else
-		RegisterSignal(parent, COMSIG_ATOM_EXIT, .proc/exit_react)
-	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/examine)
+		RegisterSignal(parent, COMSIG_ATOM_ENTERED, .proc/entered_react)
+		RegisterSignal(parent, COMSIG_ATOM_EXITED, .proc/exited_react)
+		for(var/i in parent)
+			RegisterSignal(i, COMSIG_MOVABLE_PRE_THROW, .proc/throw_react)
 
-/datum/component/magnetic_catch/proc/uncross_react(atom/movable/thing)
-	if(!thing.throwing || thing.throwing.thrower)
-		return
-	qdel(thing.throwing)
-	return COMPONENT_MOVABLE_BLOCK_UNCROSS
+/datum/component/magnetic_catch/proc/examine(mob/user)
+	to_chat(user, "It has been installed with inertia dampening to prevent coffee spills.")
 
-/datum/component/magnetic_catch/proc/exit_react(atom/movable/thing, atom/newloc)
-	if(!thing.throwing || thing.throwing.thrower)
-		return
-	qdel(thing.throwing)
-	return COMPONENT_ATOM_BLOCK_EXIT
+/datum/component/magnetic_catch/proc/crossed_react(atom/movable/thing)
+	RegisterSignal(thing, COMSIG_MOVABLE_PRE_THROW, .proc/throw_react, TRUE)
 
-/datum/component/magnetic_catch/proc/examine(mob/user)
-	to_chat(user, "It has been installed with inertia dampening to prevent coffee spills.")
\ No newline at end of file
+/datum/component/magnetic_catch/proc/uncrossed_react(atom/movable/thing)
+	UnregisterSignal(thing, COMSIG_MOVABLE_PRE_THROW)
```

Comment:
Won't this kill any `COMSIG_MOVABLE_PRE_THROW` added by other components?

Reply:
no

Reply:
`UnregisterSignal` is a proc on components that removes itself from the comp lookup list on the datum provided. That lookup list stores what component has registered for what signal and this proc will only remove itself from that registration.